client-misk/src/main/kotlin/app/cash/backfila/client/misk/internal/EmbeddedBackfillRun.kt

```diff
@@ -124,7 +124,7 @@ internal class EmbeddedBackfillRun<B : Backfill>(
       )
     when (response.batches.isEmpty()) {
       true -> cursor.done = true
-      false -> cursor.previousEndKey = response.batches.map { it.batch_range.end }.maxOrNull()
+      false -> cursor.previousEndKey = response.batches.last().batch_range.end
```

Comment:
It turns out a ByteString of '7' is > a ByteString of '10'. So we just do here what we do in the service and assume that the last batch is the the largest id.